lib/tapioca/compilers/dsl/identity_cache.rb

```diff
@@ -164,46 +166,90 @@ def create_fetch_field_methods(field, klass, returns_collection:)
         def create_fetch_by_methods(field, klass, constant)
           field_length = field.key_fields.length
           fields_name = field.key_fields.join("_and_")
+          suffix = field.send(:fetch_method_suffix)
+          is_cache_index = field.instance_variable_defined?(:@attribute_proc)
 
-          parameters = field.key_fields.map do |arg|
-            Parlour::RbiGenerator::Parameter.new(arg.to_s, type: "T.untyped")
+          # Both `cache_index` and `cache_attribute` generate aliased methods
+          create_aliased_fetch_by_methods(klass, constant, field.alias_name.to_s, field_length, suffix)
+
+          # If the method used was `cache_index` a few extra methods are created
+          if is_cache_index
+            name = "fetch_by_#{fields_name}"
+            parameters = field.key_fields.map do |arg|
+              Parlour::RbiGenerator::Parameter.new(arg.to_s, type: "T.untyped")
+            end
+            parameters << Parlour::RbiGenerator::Parameter.new("includes:", default: "nil", type: "T.untyped")
+
+            if field.unique
+              klass.create_method(
+                "#{name}!",
+                class_method: true,
+                parameters: parameters,
+                return_type: "::#{constant}"
+              )
+
+              klass.create_method(
+                name,
+                class_method: true,
+                parameters: parameters,
+                return_type: "T.nilable(::#{constant})"
+              )
+            else
+              klass.create_method(
+                name,
+                class_method: true,
+                parameters: parameters,
+                return_type: COLLECTION_TYPE.call(constant)
+              )
+            end
+
+            if field_length == 1
+              klass.create_method(
+                "fetch_multi_by_#{fields_name}",
+                class_method: true,
+                parameters: [
+                  Parlour::RbiGenerator::Parameter.new("index_values", type: "T.untyped"),
+                  Parlour::RbiGenerator::Parameter.new("includes:", default: "nil", type: "T.untyped"),
+                ],
+                return_type: COLLECTION_TYPE.call(constant)
+              )
+            end
```

Comment:
I think it would be great if this was encapsulated in a `create_index_fetch_by_methods` method.